geonode/facets/views.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from urllib.parse import urlencode
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+from rest_framework.decorators import api_view, authentication_classes
+
+from django.http import HttpResponseNotFound, JsonResponse
+from django.urls import reverse
+
+from geonode.base.models import ResourceBase
+from geonode.facets.apps import registered_facets
+from geonode.facets.models import FacetProvider, DEFAULT_FACET_TOPICS_LIMIT
+from geonode.security.utils import get_visible_resources
+
+logger = logging.getLogger(__name__)
+
+
+def list_facets(request, **kwargs):
+    lang = _resolve_language(request)
+    add_links = _resolve_boolean(request, "add_links", False)
+    include_topics = _resolve_boolean(request, "include_topics", False)
+
+    facets = []
+
+    for provider in registered_facets.values():
+        logger.debug("Fetching data from provider %r", provider)
+        info = provider.get_info(lang=lang)
+        if add_links:
+            info["link"] = f"{reverse('get_facet', args=[info['name']])}?{urlencode({'add_links': True})}"
+
+        if include_topics:
+            content = provider.get_facet_items(queryset=_prefilter_topics(request))
+            info["topics"] = content
+
+        facets.append(info)
+
+    logger.debug("Returning facets %r", facets)
+    return JsonResponse({"facets": facets})
+
+
+@api_view(["GET"])
+@authentication_classes([SessionAuthentication, BasicAuthentication])
+def get_facet(request, facet):
+    logger.debug("get_facet -> %r", facet)
+    lang = _resolve_language(request)
+    add_link = _resolve_boolean(request, "add_links", False)
+
+    provider: FacetProvider = registered_facets.get(facet)
+    if not provider:
+        return HttpResponseNotFound()
+
+    page = int(request.GET.get("page", 0))
+    limit = int(request.GET.get("limit", DEFAULT_FACET_TOPICS_LIMIT))
+
+    info = provider.get_info(lang)
+    info["user"] = request.user.username
+
+    qs = _prefilter_topics(request)
+    content = provider.get_facet_items(queryset=qs, lang=lang, page=page, limit=limit)
+
+    if add_link:
+        exist_prev = page > 0
```

Comment:
Here it will be better to use the already present pagination structure that we already use in all the other APIs (`GeoNodeApiPagination`) to keep consistency.
https://github.com/GeoNode/geonode/blob/0250ba4e85b06a28d2c01eae5d798dbf1d95aef5/geonode/base/api/pagination.py#L28-L41

Here is an example of how it is used:

https://github.com/GeoNode/geonode/blob/0250ba4e85b06a28d2c01eae5d798dbf1d95aef5/geonode/base/api/views.py#L1519-L1523